### PR TITLE
[Frontend] Adding Panning + Potential Refactoring of Zoom

### PR DESCRIPTION
**Pull request by namanhkapur**

### CODE REVIEW

File: client/src/canvas/components/Canvas.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-import { Redirect } from 'react-router-dom';
-import './Canvas.scss';
-import { Icon } from 'antd';
-import { ZOOM_CHANGE_FACTOR } from '../constants';
+import React, { Component, createRef, RefObject } from "react";
+import ColorPicker from "../../colorPicker/components/colorPicker";
+import { Redirect } from "react-router-dom";
+import "./Canvas.scss";
+import { Icon } from "antd";
+import { ZOOM_CHANGE_FACTOR } from "../constants";
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: If we're no longer using this, then I think you can remove this import?

File: client/src/canvas/components/Canvas.tsx [hunk 2]
```diff
@@ -16,49 +16,204 @@ interface Props {
...
+    this.widthCanvas = 0;
+    this.heightCanvas = 0;
+    this.xleftView = 0;
+    this.ytopView = 0;
+    this.widthViewOriginal = 1.0; //actual width and height of zoomed and panned display
+    this.heightViewOriginal = 1.0;
+    this.widthView = this.widthViewOriginal; //actual width and height of zoomed and panned display
+    this.heightView = this.heightViewOriginal;
+    this.lastX = 0;
+    this.lastY = 0;
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: does this work without the binding?

File: client/src/canvas/components/Canvas.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import React, { Component, createRef, RefObject } from 'react';
-import ColorPicker from '../../colorPicker/components/colorPicker';
-import { Redirect } from 'react-router-dom';
-import './Canvas.scss';
-import { Icon } from 'antd';
-import { ZOOM_CHANGE_FACTOR } from '../constants';
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: We need to either pick single or double quotes and stick with it.

File: client/src/canvas/components/Canvas.tsx [hunk 4]
```diff
@@ -16,49 +16,204 @@ interface Props {
...
 
-    if (context) {
-      this.props.registerContext(context);
+  draw(ctx) {
+    ctx.setTransform(1, 0, 0, 1, 0, 0);
+    ctx.scale(
+      this.widthCanvas / this.widthView,
+      this.heightCanvas / this.heightView
+    );
+    ctx.translate(-this.xleftView, -this.ytopView);
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: we need comments for lines like this

File: client/src/canvas/components/Canvas.tsx [hunk 5]
```diff
@@ -16,49 +16,204 @@ interface Props {
...
+      false
+    );
 
+    this.draw(context);
+  }
 
-    if (context) {
-      this.props.registerContext(context);
+  draw(ctx) {
+    ctx.setTransform(1, 0, 0, 1, 0, 0);
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: what does this do?

File: client/src/canvas/components/Canvas.tsx [hunk 6]
```diff
@@ -16,49 +16,204 @@ interface Props {
...
+      this.ytopView,
+      this.widthView,
+      this.heightView
+    );
+    ctx.fillStyle = "blue";
+    ctx.fillRect(0.1, 0.5, 0.1, 0.1);
+    ctx.fillStyle = "red";
+    ctx.fillRect(0.3, 0.2, 0.4, 0.2);
+    ctx.fillStyle = "green";
+    ctx.beginPath();
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: I think you could go ahead and get this to use the actual color that we choose and not these hard-coded ones as that code has been merged in.

File: client/src/canvas/components/Canvas.tsx [hunk 7]
```diff
@@ -73,21 +228,24 @@ class Canvas extends Component<Props> {
...
-      <div className='canvas-container'>
+      <div className="canvas-container">
         {/* <ColorPicker
           onCancel={() => console.log('canceled')}
           onComplete={(c) => console.log('color selected: ', c)}
         /> */}
-        <div className='zoom-canvas' style={{ transform: `scale(${zoomFactor}, ${zoomFactor})` }}>
+        <div
+          className="zoom-canvas"
+          //   style={{ transform: `scale(${zoomFactor}, ${zoomFactor})` }}
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: if we're doing this then you should just delete the style={...

File: client/src/canvas/components/Canvas.tsx [hunk 8]
```diff
@@ -16,49 +16,204 @@ interface Props {
...
+    var y = (Y / this.heightCanvas) * this.heightView + this.ytopView;
+
+    var scale = event.shiftKey == 1 ? 1.5 : 0.5; // shrink (1.5) if shift key pressed
+    this.widthView *= scale;
+    this.heightView *= scale;
+
+    if (
+      this.widthView > this.widthViewOriginal ||
+      this.heightView > this.heightViewOriginal
+    ) {
```
Comment (almagee16) [inline client/src/canvas/components/Canvas.tsx]: nit: this looks weird. i feel like it is fine to put this on the same line.

**Comment by namanhkapur** [comment]
archaic

**Pull request closed without merging**